providers/base/utils.go

```diff
@@ -0,0 +1,43 @@
+package base
+
+import (
+	providertypes "github.com/skip-mev/slinky/providers/types"
+)
+
+type (
+	// The ConfigUpdater is an interface that be utilized to fetch the configuration
+	// for a given provider. This allows the provider to be updated asynchronously.
+	ConfigUpdater[K providertypes.ResponseKey] interface {
+		// GetIDs is the channel that is used to update the set of IDs that the provider
+		// will fetch data for. This blocks until there is a viable update.
```

Comment:
Any reason to choose a channel v. a RWMutex here? Don't you want all Providers to read the updated ids concurrently? Do we expect a ConfigUpdater per Provider?

Reply:
Right now, this is being done per provider

Reply:
yea the interface that the config client, which is the client that is responsible for updating all providers, will use is the ConfigUpdater - which is specific per provider.